### PR TITLE
Sawfly bugfixes

### LINKED ISSUE #10105: Sawflies can cut through solid matter from about 10 tiles away
### Labels

[BUG]

### Description

Sawflies apparently ignore distance and accessibility and cut you regardless

### Steps to reproduce

1. Build a small room in escape
2. Have someone deploy sawflies in escape
3. Be killed though the windows of your room
4. Make sure there is no drone in with your room
5. Pain

### Expected Behavior

To be safe in a secluded area from melee based attacks

### Additional Information & Screenshots

unknown
unknown

### LINKED ISSUE #10240: sawflies have a move slowdown on crit/damage
### Labels

[BUG]

### Description

I'm unsure if this is a bug or intentional, but it doesn't make sense for mob/robotic s to have the same injured slowdown penalty.

### Steps to reproduce

1. Be a sawfly
2. Be in crit
3. Be slowed by robo-pain

### Expected Behavior

I'm pretty sure this is recent, but I don't think robotic things shouldn't have the slowdown organics do

### Additional Information & Screenshots
This could very well be a feature for mob/livings, but it's detrimental to sawflies as a traitor item to have this behavior.

### PR DESCRIPTION
**Pull request by NightmareChamillian**
[bug]
## About the PR 
Makes Sawflies have more aggressive range checks on their attacks, and fixes them having an injury slowdown. If the latter is problematic let me know but I believe it is an oversight or a bug.
## Why's this needed? 
Bugs no good
Closes #10240
Closes #10105

### CODE REVIEW

**Comment by NightmareChamillian** [comment]
oh NO it's cursed

**Review by ZeWaka — CHANGES_REQUESTED** [review]
fix ur pr desc

File: code/mob/living.dm [hunk 1]
```diff
@@ -1666,13 +1666,14 @@ var/global/icon/human_static_base_idiocy_bullshit_crap = icon('icons/mob/human.d
 		return .
 
 	var/health_deficiency = 0
-	if (src.max_health > 0)
-		health_deficiency = ((src.max_health-src.health)/src.max_health)*100 + health_deficiency_adjustment // cogwerks // let's treat this like pain
-	else
-		health_deficiency = (src.max_health-src.health) + health_deficiency_adjustment
+	if(!istype(src, /mob/living/critter/robotic/sawfly)) //disables pain slowdown for sawflies- consider upgrading to all robots later
```
Comment (Tarmunora) [inline code/mob/living.dm:1669]: no. bad.
Just give sawflies a movement modifier with -INFINITY health_deficiency_adjustment or something

File: code/mob/living/critter/ai/sawflyai.dm [hunk 2]
```diff
@@ -68,12 +68,13 @@
 			if(src.found_path)
 				walk_rand(src,4)
 				holder.move_to_with_path(holder.target, src.found_path, 1)
-				owncritter.set_dir(get_dir(owncritter, holder.target)) //attack regardless
-				owncritter.hand_attack(holder.target, dummy_params)
-			//frustration++ //if frustration gets too high, the task is ended and re-evaluated
+				if(GET_DIST(owncritter, holder.target) > target_range)
```
Comment (Tarmunora) [inline code/mob/living/critter/ai/sawflyai.dm:71]: use `in_interact_range()` for this kind of melee range check

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/mob/living/critter/sawfly.dm [hunk 3]
```diff
@@ -29,6 +29,7 @@ This file is the critter itself, and all the custom procs it needs in order to f
 	var/list/dummy_params = list("icon-x" = 16, "icon-y" = 16) //for the manual attack_hand retaliation
 
 	//mob variables
+	health_deficiency_adjustment = -200000 //prevents them from having movespeed slowdown when injured
```
Comment (ZeWaka) [inline code/mob/living/critter/sawfly.dm:32]: Please just use actual infinity

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Pull request merged**